escape.go

```diff
@@ -191,39 +191,59 @@ func (ei *escapeInterpreter) output256() error {
 		return ei.outputNormal()
 	}
 
-	fgbg, err := strconv.Atoi(ei.csiParam[0])
-	if err != nil {
-		return errCSIParseError
-	}
-	color, err := strconv.Atoi(ei.csiParam[2])
-	if err != nil {
-		return errCSIParseError
-	}
-
-	switch fgbg {
-	case 38:
-		ei.curFgColor = Attribute(color + 1)
+	for _, param := range ei.splitFgBg() {
+		fgbg, err := strconv.Atoi(param[0])
+		if err != nil {
+			return errCSIParseError
+		}
+		color, err := strconv.Atoi(param[2])
+		if err != nil {
+			return errCSIParseError
+		}
 
-		for _, param := range ei.csiParam[3:] {
-			p, err := strconv.Atoi(param)
-			if err != nil {
-				return errCSIParseError
+		switch fgbg {
+		case 38:
+			ei.curFgColor = Attribute(color + 1)
+
+			for _, s := range param[3:] {
+				p, err := strconv.Atoi(s)
+				if err != nil {
+					return errCSIParseError
+				}
+
+				switch {
+				case p == 1:
+					ei.curFgColor |= AttrBold
+				case p == 4:
+					ei.curFgColor |= AttrUnderline
+				case p == 7:
+					ei.curFgColor |= AttrReverse
+
+				}
 			}
+		case 48:
+			ei.curBgColor = Attribute(color + 1)
+		default:
+			return errCSIParseError
+		}
+	}
+	return nil
+}
 
-			switch {
-			case p == 1:
-				ei.curFgColor |= AttrBold
-			case p == 4:
-				ei.curFgColor |= AttrUnderline
-			case p == 7:
-				ei.curFgColor |= AttrReverse
-			}
+func (ei *escapeInterpreter) splitFgBg() [][]string {
```

Comment:
does this function need to be a method of `ei`, cant this just be a function we can call like 
```
func splitFgBg() [][]string { ... return out}
```

Reply:
spltFfBg is very specific to the escapeInterpreter, so to me it makes sense to keep it as is.

Reply:
I get that, but it does not rely on ei, it just relies on data. It is not driven by the struct

Reply:
Not a big deal to me.  I'll change it.  Just a sec.

Reply:
done